src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -225,13 +225,12 @@ prepare_download_pkgs_txn(const gchar * const *queries,
     return glnx_prefix_error (error, "Downloading metadata");
 
   DnfSack *sack = dnf_context_get_sack (rpmostree_context_get_dnf (ctx));
-  RpmOstreeOverrideSourceKind source_kind;
-  g_autofree char * source_name = rpmostree_parse_override_source (source, &source_kind, error);
-  if (source_name == NULL)
-    return glnx_prefix_error (error, "Parsing override source kind");
+  auto parsed_source = CXX_TRY_VAL(parse_override_source(source), error);
```

Comment:
Unfortunately I didn't see a way to return an heterogeneous tuple `(kindEnum, String)` from Rust to C++ here, so I had to resort to duplicate calls for splitting and re-parsing. Not the best, but not a big deal either.

Reply:
Wouldn't a struct work?

Reply:
Yes, my gut feeling was that it was a bit overkill, but overall it may indeed be cleaner.
Would you prefer that? I'll be happy to rework it in that direction if so.

Reply:
This is a great micro-example of the eternal tension of: we have code that is better and was probably tested locally, is passing CI, but could be further improved - is it worth re-working it again, paying the cost of re-review and CI and the risk of adding new bugs?

I don't have a strong opinion on this particular case :smile: - marked approved, feel free to merge as is or tweak!

Reply:
This is not in the blocking path for anything else, and I don't plan to touch it again soon. So I think it is worth to address the comment now. I've added a separate comment on top.